src/hooks/useRankingData.js

```diff
@@ -54,14 +71,44 @@ const useRankingData = () => {
     })();
   }, [today]);
 
+  useEffect(() => {
+    (async function () {
+      const q = query(collection(firebaseDb, USERS_STATISTICS), orderBy('currentStreak', 'desc'));
+      const docs = await getDocs(q);
+      const results = [];
+      let position = 0;
+      let currentStreakValue = Number.MAX_SAFE_INTEGER;
+      docs.forEach(async doc => {
+        const { id: userEmail } = doc;
+        const { currentStreak, lastDatePlayed, totalGames } = doc.data();
+        if (!!totalGames) {
+          if (currentStreak !== currentStreakValue) {
+            currentStreakValue = currentStreak;
+            position += 1;
+          }
+          results.push({
+            currentStreak,
+            lastDatePlayed,
+            position,
+            user: users[userEmail] || { email: userEmail, name: userEmail.split('@')[0] },
+          });
+        }
+      });
+      setCurrentStreakRankingData(results);
+      setLoading(false);
+    })();
+  }, [users]);
+
   const currentUserPlayed = dailyResults.find(item => item.user.email === currentUser);
```

Comment:
We can memorize the value of currentUserPlayed and avoid recreating it on every re-render.